tools/perf/ResultsComparer/src/ResultsComparer/Core/Reporting/MarkdownReporter.cs

```diff
@@ -0,0 +1,144 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MarkdownReporter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using MarkdownLog;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace ResultsComparer.Core.Reporting
+{
+    /// <summary>
+    /// Generates comparison reports in markdown format.
+    /// </summary>
+    public class MarkdownReporter : IReporter
+    {
+        /// <inheritdoc/>
+        public void GenerateReport(ComparerResults results, Stream destination, ComparerOptions args)

```

Comment:
Probably better to use `ComparerOptions options` (just like in the interface) instead of `ComparerOptions args`